src/common/downloadFileDialog.component.tsx

```diff
@@ -0,0 +1,110 @@
+import WarningIcon from '@mui/icons-material/Warning';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormHelperText,
+} from '@mui/material';
+import { useQueryClient } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import React from 'react';
+import { APIImage, APIImageWithURL } from '../api/api.types';
+import { getImageQuery } from '../api/images';
+import handleIMS_APIError from '../handleIMS_APIError';
+import { downloadFileByLink } from '../utils';
+
+export interface BaseDownloadFileProps {
+  open: boolean;
+  onClose: () => void;
+  fileType: 'Image' | 'Attachment';
+}
+
+export interface ImageDownloadDialogProps extends BaseDownloadFileProps {
+  fileType: 'Image';
+  file: APIImage;
+}
+
+export type DownloadFileProps = ImageDownloadDialogProps;
+
+const DownloadFileDialog = (props: DownloadFileProps) => {
+  const { open, onClose, fileType, file } = props;
+
+  const getFile = fileType === 'Image' ? getImageQuery : getImageQuery;
+
+  const [formError, setFormError] = React.useState<string | undefined>(
+    undefined
+  );
+
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  const queryClient = useQueryClient();
+
+  const handleClose = React.useCallback(() => {
+    setFormError(undefined);
+    setIsLoading(false);
+    onClose();
+  }, [onClose]);
+
+  const handleClick = React.useCallback(async () => {
+    setIsLoading(true);
+    queryClient
+      .fetchQuery(getFile(file.id, false))
+      .then((data: APIImageWithURL) => {
+        setIsLoading(false);
+        downloadFileByLink(document, data.download_url, data.file_name);
+        onClose();
+      })
+      .catch((error: AxiosError) => {
+        setIsLoading(false);
+        handleIMS_APIError(error);
+        setFormError('No data provided. Please refresh and try again');
```

Comment:
Remove this error message as it is not needed. It was used when the `selectedData` prop was optional, but that is not the case for this dialog. This could also lead to confusion, as the error message would not match the toast error in some instances.